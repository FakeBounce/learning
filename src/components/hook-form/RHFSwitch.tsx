import { memo, ReactNode } from 'react';
import { Box, FormControlLabel, Typography } from '@mui/material';
import LMSSwitch from '@src/components/lms/LMSSwitch';
import { Controller, useFormContext } from 'react-hook-form';
import LabelWithRequired from '@src/components/hook-form/LabelWithRequired';

interface RHFSwitchProps {
  name: string;
  required?: boolean;
  label: ReactNode;
  disabled?: boolean;
}

function RHFSwitch({ name, required = false, label, disabled }: RHFSwitchProps) {
  const { control } = useFormContext();
  return (
    <Box>
      <Typography variant="caption">
        {required ? <LabelWithRequired name={name} label={label} /> : label}
      </Typography>
      <Box ml={1.5} mt={1}>
        <Controller
          name={name}
          control={control}
          defaultValue={control._defaultValues[name]}
          render={({ field: { onChange, value } }) => {
            return (
              <FormControlLabel
                control={<LMSSwitch checked={value} handleChange={onChange} disabled={disabled} />}
                label=""
              />
            );
          }}
        />
      </Box>
    </Box>
  );
}

export default memo(RHFSwitch);
