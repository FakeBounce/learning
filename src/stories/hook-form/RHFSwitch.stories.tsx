import { Meta, StoryObj } from '@storybook/react';
import { FormProvider, useForm } from 'react-hook-form';
import RHFSwitch from '@src/components/hook-form/RHFSwitch';

const meta: Meta = {
  title: 'Example/RHFSwitch',
  component: RHFSwitch,
  parameters: {
    layout: 'centered'
  }
};

export default meta;

type Story = StoryObj<typeof RHFSwitch>;

const RHFSwitchTemplate: Story = {
  render: (args: any) => {
    const methods = useForm();

    return (
      <FormProvider {...methods}>
        <form>
          <RHFSwitch {...args} />
        </form>
      </FormProvider>
    );
  }
};

export const Default = {
  ...RHFSwitchTemplate,
  args: {
    name: 'default',
    label: 'Default Switch',
    required: false
  }
};

export const Required = {
  ...RHFSwitchTemplate,
  args: {
    name: 'required',
    label: 'Required',
    required: true
  }
};

export const Checked = {
  ...RHFSwitchTemplate,
  args: {
    name: 'checked',
    label: 'Checked Switch',
    required: false,
    defaultValue: true
  }
};
