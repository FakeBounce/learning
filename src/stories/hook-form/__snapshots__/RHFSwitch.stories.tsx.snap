// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/RHFSwitch Checked smoke-test 1`] = `
<form>
  <div class="MuiBox-root css-0">
    Checked Switch
    <div class="MuiBox-root css-1ldhbnq">
      <label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1vbggne-MuiFormControlLabel-root">
        <span class="MuiSwitch-root MuiSwitch-sizeMedium css-tvdqvb-MuiSwitch-root">
          <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1brz8fh-MuiButtonBase-root-MuiSwitch-switchBase">
            <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                   type="checkbox"
            >
            <span class="MuiSwitch-thumb css-tqmoez-MuiSwitch-thumb">
            </span>
            <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
            </span>
          </span>
          <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
          </span>
        </span>
        <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-gkbk35-MuiTypography-root">
        </span>
      </label>
    </div>
  </div>
</form>
`;

exports[`Example/RHFSwitch Default smoke-test 1`] = `
<form>
  <div class="MuiBox-root css-0">
    Default Switch
    <div class="MuiBox-root css-1ldhbnq">
      <label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1vbggne-MuiFormControlLabel-root">
        <span class="MuiSwitch-root MuiSwitch-sizeMedium css-tvdqvb-MuiSwitch-root">
          <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1brz8fh-MuiButtonBase-root-MuiSwitch-switchBase">
            <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                   type="checkbox"
            >
            <span class="MuiSwitch-thumb css-tqmoez-MuiSwitch-thumb">
            </span>
            <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
            </span>
          </span>
          <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
          </span>
        </span>
        <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-gkbk35-MuiTypography-root">
        </span>
      </label>
    </div>
  </div>
</form>
`;

exports[`Example/RHFSwitch Required smoke-test 1`] = `
<form>
  <div class="MuiBox-root css-0">
    Required
    <span class="MuiFormLabel-asterisk">
      *
    </span>
    <div class="MuiBox-root css-1ldhbnq">
      <label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1vbggne-MuiFormControlLabel-root">
        <span class="MuiSwitch-root MuiSwitch-sizeMedium css-tvdqvb-MuiSwitch-root">
          <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1brz8fh-MuiButtonBase-root-MuiSwitch-switchBase">
            <input class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                   type="checkbox"
            >
            <span class="MuiSwitch-thumb css-tqmoez-MuiSwitch-thumb">
            </span>
            <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
            </span>
          </span>
          <span class="MuiSwitch-track css-1yjjitx-MuiSwitch-track">
          </span>
        </span>
        <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-gkbk35-MuiTypography-root">
        </span>
      </label>
    </div>
  </div>
</form>
`;
